network/dag/transaction.go

```diff
@@ -114,7 +117,8 @@ type Transaction interface {
 
 // NewTransaction creates a new unsigned transaction. Parameters payload and payloadType can't be empty, but prevs is optional.
 // Prevs must not contain empty or invalid hashes. Duplicate prevs will be removed when given.
-func NewTransaction(payload hash.SHA256Hash, payloadType string, prevs []hash.SHA256Hash) (UnsignedTransaction, error) {
+// The pal byte slice (may be nil) holds the encrypted recipient address, if it is a private transaction.
+func NewTransaction(payload hash.SHA256Hash, payloadType string, prevs []hash.SHA256Hash, pal [][]byte) (UnsignedTransaction, error) {
```

Comment:
custom type for `[][]byte` and some convenience methods to transform []did to that?